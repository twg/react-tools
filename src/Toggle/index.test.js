/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { Toggle } from './'

const noop = () => {}
describe('Toggle', () => {
  it('renders an untoggled toggle', () => {
    let component = renderer.create(
      <Toggle
        onChange={noop}
        leftLabel="Unchecked Example"
        rightLabel="Unchecked Example"
      />
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('renders a toggled toggle', () => {
    let component = renderer.create(
      <Toggle
        onChange={noop}
        leftLabel="Unchecked Example"
        rightLabel="Unchecked Example"
        initialValue={true}
      />
    )

    expect(component.toJSON()).toMatchSnapshot()
  })
})
