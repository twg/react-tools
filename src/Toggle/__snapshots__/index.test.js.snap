// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle renders a toggled toggle 1`] = `
<div
  className="toggleSwitchWrapper"
>
  <label
    className="beforeLabel"
  >
    Unchecked Example
  </label>
  <div
    className="toggle"
    onClick={[Function]}
  >
    <div
      className="toggleBall"
    />
  </div>
  <label
    className="afterLabel"
  >
    Unchecked Example
  </label>
</div>
`;

exports[`Toggle renders an untoggled toggle 1`] = `
<div
  className="toggleSwitchWrapper"
>
  <label
    className="beforeLabel"
  >
    Unchecked Example
  </label>
  <div
    className="toggle"
    onClick={[Function]}
  >
    <div
      className="toggleBall"
    />
  </div>
  <label
    className="afterLabel"
  >
    Unchecked Example
  </label>
</div>
`;
